File: maddie-carlson/day-06/coin-combos.js
function coinCombo(cents) {
  // needs to return an array with the least amount of coins, going from pennies to nickels to dimes to quarters
  // to accomplish this, we need to count in reverse, largest coins first, then smallest coins
  // this can be done by using while loops and gradually subtracting from our cents until the current coin is too large
  let coins = [0, 0, 0, 0];

  // quarters
  while (cents >= 25) {
    coins[3] += 1
    cents -= 25
  }

  // dimes
  while (cents >= 10) {
    coins[2] += 1
    cents -= 10
  }

  // nickels
  while (cents >= 5) {
    coins[1] += 1
    cents -= 5
  }

  // pennies
  coins[0] = cents // cover the remainder

  return coins;
}
/*
  Ember's Feedback:
  - Great job! Wonderful pseudo code and great solution
*/
// Alternative solution
function coinCombo(cents) {
  let coins = [25, 10, 5, 1]
  let remainder = cents
  return coins.map(coin => {
    let coinsInCents = Math.floor(remainder / coin)
    remainder %= coin
    return coinsInCents
  }).reverse();
}

File: maddie-carlson/day-29/coin-combos.py
import math


def coin_combo(cents):
    # function takes an amount of cents and needs to return it divided into an array of coins
    # [pennies, nickels, dimes, quarters]

    # to solve, init our array and then go down the list, resolving the coins largest to smallest
    coins = [0, 0, 0, 0]

    # go coin by coin, dividing and then setting our remaining coins to the remainder
    if cents >= 25:
        coins[3] = math.floor(cents / 25)
        cents = cents % 25

    if cents >= 10:
        coins[2] = math.floor(cents / 10)
        cents = cents % 10

    if cents >= 5:
        coins[1] = math.floor(cents / 5)
        cents = cents % 5

    # remaining coins are our pennies
    coins[0] = cents

    return coins

"""
    Ember's Feedback:
    - Awesome job :)
"""
# Alternative Solution


def coin_combo(cents):
    quarters = cents // 25
    dimes = cents % 25 // 10
    nickels = cents % 10 // 5
    pennies = cents % 5
    return [pennies, nickels, dimes, quarters]
